### DISCUSSION TITLE
Extract JPEG Exif Data

### DISCUSSION THREAD

**Discussion by schall1337** [pymupdf/PyMuPDF#1953] [Looking for help] [answered]
Hi,

how can I save JPEG Images from PDF and keeping the EXIF data (author, copyright etc.)?

```
# extract the image bytes
base_image = pdf_file.extract_image(xref)
image_bytes = base_image["image"]
# get the image extension
image_ext = base_image["ext"]
# load it to PIL
image = Image.open(io.BytesIO(image_bytes))

# save it to local disk
image.save(open(f"image{page_index+1}_{image_index}.{image_ext}", "wb"))
```

Saves the file correctly, but the EXIF data is missing. 
I am sure the image has some EXIF data as other tools like pikepdf extract the image correctly with the exif data?

**Comment by schall1337** [comment]
got it, its more a pillow thing

get exif with

`exif = image.getexif()`

and then save like this

`image.save(open(f"image{page_index+1}_{image_index}.{image_ext}", "wb"), exif = exif)`

**Comment by JorjMcKie** [comment, marked as answer]
Actually, if you do this, the data stored in the PDF should be saved to disk 1:1 - i.e. including any EXIF:
```python
img = doc.extract_image(xref)
out = open(f"{xref}.{img['ext']}", "wb")
out.write(img["image"])
out.close()
```

**Reply by schall1337** [reply]
you're right. Thanks!!
